src/lib/bootstrap_controller/bootstrap_controller.ml

```diff
@@ -273,6 +306,41 @@ end = struct
       let root_sync_ledger =
         Root_sync_ledger.create ledger_db ~logger:t.logger ~trust_system
       in
+      (let%bind () = Ivar.read ask_best_tip_signal in
```

Comment:
Why do you actually need this synchronization? You are filling this ivar once you get the first connection, but nothing else in the bootstrap controller will react to anything until it receives information over the network, which is after the first connection, right? Why can't you just synchronize outside before calling `run` in the bootstrap controller?